internal/resources/asserts/resource_custom_model_rules.go

```diff
@@ -0,0 +1,478 @@
+
+		entityMap := map[string]interface{}{
+			"type":       entityType,
+			"name":       entityName,
+			"defined_by": definedBy,
+		}
+
+		// Add optional entity fields if they exist
+		if len(entity.Scope) > 0 {
+			entityMap["scope"] = entity.Scope
+		}
+
+		if len(entity.Lookup) > 0 {
+			entityMap["lookup"] = entity.Lookup
+		}
+
+		if len(entity.EnrichedBy) > 0 {
+			var enrichedByList []string
+			for _, enrichedBy := range entity.EnrichedBy {
+				if enrichedBy.Query != nil {
+					enrichedByList = append(enrichedByList, *enrichedBy.Query)
+				}
+			}
+			if len(enrichedByList) > 0 {
+				entityMap["enriched_by"] = enrichedByList
+			}
+		}
+
+		if entity.Disabled != nil {
+			entityMap["disabled"] = *entity.Disabled
+		}
+
+		entities = append(entities, entityMap)
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"entity": entities,
+		},
+	}, nil
+}
+
+func makeResourceCustomModelRules() *common.Resource {
+	sch := &schema.Resource{
+		Description: "Manages Asserts Custom Model Rules through the Grafana API.",
+
+		CreateContext: resourceCustomModelRulesCreate,
+		ReadContext:   resourceCustomModelRulesRead,
+		UpdateContext: resourceCustomModelRulesUpdate,
+		DeleteContext: resourceCustomModelRulesDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name of the custom model rules.",
+			},
+			"rules": {
+				Type:        schema.TypeList,
+				Required:    true,
+				MaxItems:    1,
+				Description: "The rules configuration for the custom model rules.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"entity": {
+							Type:        schema.TypeList,
+							Required:    true,
+							Description: "List of entities to define in the custom model rules.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"type": {
+										Type:        schema.TypeString,
+										Required:    true,
+										Description: "The type of the entity (e.g., Service, Pod, Namespace).",
+									},
+									"name": {
+										Type:        schema.TypeString,
+										Required:    true,
+										Description: "The name of the entity.",
+									},
+									"scope": {
+										Type:        schema.TypeMap,
+										Optional:    true,
+										Description: "Scope labels for the entity.",
+										Elem:        &schema.Schema{Type: schema.TypeString},
+									},
+									"lookup": {
+										Type:        schema.TypeMap,
+										Optional:    true,
+										Description: "Lookup mappings for the entity.",
+										Elem:        &schema.Schema{Type: schema.TypeString},
+									},
+									"enriched_by": {
+										Type:        schema.TypeList,
+										Optional:    true,
+										Description: "List of enrichment sources for the entity.",
```

Comment:
I think this should be similar to `defined_by`.
Something like:
`List of queries that enrich entities of this type.`